### PR TITLE
[UDF] Add UDF for emulating snowflake split_part

### PR DESCRIPTION
**Pull request by bdw-g**
This adds a function for emulating snowflake split_part. This function has the following semantics:

- The part index is 1-based
- However, index 0 is treated as 1
- Negative numbers are interpreted as indexes from the end of the string

It will help customers migrate from Snowflake to BigQuery.

### CODE REVIEW

File: udfs/community/test_cases.js [hunk 1]
```diff
@@ -3009,7 +3009,72 @@ generate_udf_test("cw_period_rdiff", [
         expected_output: `NULL`
     },
 ]);
-
+generate_udf_test("cw_split_part_delimstr_idx", [
```
Comment (danieldeleo) [inline udfs/community/test_cases.js:3012]: Please add test case for NULL inputs

**Review by danieldeleo — APPROVED** [review]
Will merge after null test cases are added, thanks!

File: udfs/community/test_cases.js (hunk 1 above)
Comment (bdw-g) [reply udfs/community/test_cases.js:3012]: Done.
